### PR TITLE
FriendlyURLGenerator: generateFromDashboardName method fixed

### PR DESCRIPTION
**Pull request by denisquesada**
## Description

`generateFromDashboardName` method code refactor

## Testing

Describe how you tested your changes and/or give instructions to the reviewers on how they can verify them.

`npm test`

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/src/lib/services/friendlyurl-service.ts [hunk 1]
```diff
@@ -43,12 +43,11 @@ function isValid(friendlyURL: string): boolean {
 
 function generateFriendlyURL(dashboardName: string): string {
   return dashboardName
-    .trim()
     .toLocaleLowerCase()
     .replace(/[!#$&'\(\)\*\+,\/:;=\?@\[\]]+/g, " ") // remove RFC-3986 reserved characters
+    .trim() // remove spaces from the beginning and the end.
     .replace(/\s+/g, "-") // replace spaces for dashes
-    .replace(/-+/g, "-") // convert consecutive dashes to singular dash
-    .replace(/^(-)+|(-)+$/g, ""); // remove dashes at the end and beginning
```
Comment (mabreuortega) [inline backend/src/lib/services/friendlyurl-service.ts:51]: what about dashes at the beginning or the end

```
"-test-name-"
```

File: frontend/src/services/FriendlyURLGenerator.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
 function generateFromDashboardName(dashboardName: string): string {
   return dashboardName
-    .trim()
     .toLocaleLowerCase()
     .replace(/[!#$&'()*+,/:;=?@[\]]+/g, " ") // remove RFC-3986 reserved characters
+    .trim() // remove spaces from the beginning and the end.
     .replace(/\s+/g, "-") // replace spaces for dashes
-    .replace(/-+/g, "-") // convert consecutive dashes to singular dash
-    .replace(/^(-)+|(-)+$/g, ""); // remove dashes at the end and beginning
```
Comment (mabreuortega) [inline frontend/src/services/FriendlyURLGenerator.ts:8]: ditto

**Review by mabreuortega — COMMENTED** [review]
Question asked

File: backend/src/lib/services/friendlyurl-service.ts (hunk 1 above)
Comment (denisquesada) [reply backend/src/lib/services/friendlyurl-service.ts:51]: the stated test case was added to the method as well to the unit tests.

**Review by mabreuortega — APPROVED** [review]
LGTM

**Pull request merged**